lambda/audit.py

```diff
@@ -0,0 +1,319 @@
+    audit_id = str(uuid.uuid4())
+    LOG.info({"action": "Starting github audit", "org": org, "audit_id": audit_id})
+
+    sns_org_members = create_sns_message(
+        {"action": "log_org_membership", "org": org, "audit_id": audit_id}
+    )
+    publish_alert(sns_org_members)
+
+    sns_org_repos = create_sns_message(
+        {"action": "log_org_repos", "org": org, "audit_id": audit_id}
+    )
+    publish_alert(sns_org_repos)
+
+    sns_org_teams = create_sns_message(
+        {"action": "log_org_teams", "org": org, "audit_id": audit_id}
+    )
+    publish_alert(sns_org_teams)
+    return True
+
+
+def log_org_membership(message=None):
+    """ Audit github organization membership """
+    org = os.environ.get("GITHUB_ORG")
+    audit_id = message.get("audit_id")
+    if audit_id is not None:
+        LOG.info({"action": "Audit organization members", "org": org})
+        members = github_api.get_github_org_members(org)
+        event = {
+            "type": "OrganizationMemberCount",
+            "org": org,
+            "count": len(members),
+            "audit_id": audit_id,
+        }
+        LOG.info(event)
+        for member in members:
+            event = {
+                "type": "OrganizationMember",
+                "org": org,
+                "member": member,
+                "team": None,
+                "repository": None,
+                "audit_id": audit_id,
+            }
+            LOG.info(event)
+    return True
+
+
+def log_org_teams(message=None):
+    """ Audit github organization teams """
+    org = os.environ.get("GITHUB_ORG")
+    audit_id = message.get("audit_id")
+    if audit_id is not None:
+        LOG.info({"action": "Audit organization teams", "org": org})
+        teams = github_api.get_github_org_teams(org)
+        event = {
+            "type": "OrganizationTeamCount",
+            "org": org,
+            "count": len(teams),
+            "audit_id": audit_id,
+        }
+        LOG.info(event)
+        for team in teams:
+            event = {
+                "type": "OrganizationTeam",
+                "org": org,
+                "member": None,
+                "team": team,
+                "repository": None,
+                "audit_id": audit_id,
+            }
+            LOG.info(event)
+            sns_team_membership = create_sns_message(
+                {
+                    "action": "log_org_team_membership",
+                    "org": org,
+                    "team": team,
+                    "audit_id": audit_id,
+                }
+            )
+            publish_alert(sns_team_membership)
+
+            sns_team_repos = create_sns_message(
+                {
+                    "action": "log_org_team_repos",
+                    "org": org,
+                    "team": team,
+                    "audit_id": audit_id,
+                }
+            )
+            publish_alert(sns_team_repos)
+    return True
+
+
+def log_org_team_membership(message=None):
```

Comment:
> Is this default value of `None` a good idea? You use the value without a null check which would break straight away.

That's a good point. I think originally the `None` was to allow for a lambda invoke from cron with no payload but I don't think we need that anymore since the cron'd invokes now have a payload to call the different actions.